webapps/docs/config/http.xml

```diff
@@ -208,6 +208,24 @@
       The default is <code>POST</code></p>
     </attribute>
 
+    <attribute name="path" required="false">
```

Comment:
At the moment the `port` parameter is mandatory, i.e. if `path` is specified then both TCP and UDS will be exposed.
I guess there will be users which will want to enable **only** UDS.
Do you think this is reasonable and we should make it possible to disable TCP in case UDS is enabled ?

Reply:
Having both enabled might lead to confusion as it does now with dual-stack socket. Most don't even know that one can pass sockopt IPV6_ONLY. I would avoid such situation unless the admin does exactly know what is happening here. Now the misfortune here is that we don't have an address field, but rather a host/port which is tied to TCP sockets. Maybe it is worth a discussion on the dev ML to move to `address` instead of `host`/`port` combination. So do all decent socket-based services like sshd, httpd, etc. WDYT?

Reply:
I've just tested it. It seems only UDS is enabled:

```
openjdk version "16-ea" 2021-03-16
OpenJDK Runtime Environment (build 16-ea+26-1764)
OpenJDK 64-Bit Server VM (build 16-ea+26-1764, mixed mode)
=== Protocol: org.apache.coyote.http11.Http11AprProtocol
=== Starting : Connector[org.apache.coyote.http11.Http11AprProtocol-8080]
Dec 11, 2020 11:55:52 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
INFO: Loaded Apache Tomcat Native library [1.2.27] using APR version [1.6.5].
Dec 11, 2020 11:55:52 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
INFO: APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true], UDS [true].
Dec 11, 2020 11:55:52 AM org.apache.catalina.core.AprLifecycleListener initializeSSL
INFO: OpenSSL successfully initialized [OpenSSL 1.1.1g-dev  xx XXX xxxx]
Dec 11, 2020 11:55:52 AM org.apache.coyote.http11.AbstractHttp11Protocol configureUpgradeProtocol
INFO: The ["https-openssl-apr-/tmp/tomcat-uds.sock"] connector has been configured to support negotiation to [h2] via ALPN
Dec 11, 2020 11:55:52 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["https-openssl-apr-/tmp/tomcat-uds.sock"]
Dec 11, 2020 11:55:52 AM org.apache.catalina.core.StandardService startInternal
INFO: Starting service [Tomcat]
Dec 11, 2020 11:55:52 AM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet engine: [Apache Tomcat/10.0.0-M11-dev]
Dec 11, 2020 11:55:53 AM org.apache.coyote.AbstractProtocol start
INFO: Starting ProtocolHandler ["https-openssl-apr-/tmp/tomcat-uds.sock"]
=== Started

```

```
$  sudo netstat -anp | grep 2887991
(standard input):35:unix  2      [ ACC ]     STREAM     LISTENING     14491510 2887991/java         /tmp/tomcat-uds.sock
(standard input):165:unix  2      [ ]         STREAM     CONNECTED     14491508 2887991/java     
```